File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/cpp/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.cpp
#include "CreatureService.h"

namespace de::sots::cellarsandcentaurs::domain::service {

CreatureService::CreatureService(std::shared_ptr<Creatures> creatures)
    : creatures_{std::move(creatures)} {}

void CreatureService::save(const model::Creature& creature) {
    creatures_->save(creature);
}

}

File: sample-projects/typescript/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.ts
import { createLogger, transports, Logger } from "winston";
import { Creatures } from "./Creatures";
import { Creature } from "../model/Creature";

export class CreatureService {
    private readonly logger: Logger = createLogger({ transports: [new transports.Console()] });

    constructor(private readonly creatures: Creatures) {}

    save(creature: Creature): void {
        this.logger.info(`saving creature ${creature.getId().id}`);
        this.creatures.save(creature);
    }
}
